src/BlockTypes/AddToCartForm.php

```diff
@@ -73,6 +105,45 @@ protected function render( $attributes, $content, $block ) {
 		return $form;
 	}
 
+	/**
+	 * Add a hidden input to the Add to Cart form to indicate that it is a descendent of a Single Product block.
+	 *
+	 * @param string $product The Add to Cart Form HTML.
+	 * @param string $is_descendent_of_single_product_block Indicates if block is descendent of Single Product block.
+	 *
+	 * @return string The Add to Cart Form HTML with the hidden input.
+	 */
+	protected function add_is_descendent_of_single_product_block_hidden_input_to_product_form( $product, $is_descendent_of_single_product_block ) {
+
+		$hidden_is_descendent_of_single_product_block_input = sprintf(
+			'<input type="hidden" name="is-descendent-of-single-product-block" value="%1$s">',
+			$is_descendent_of_single_product_block ? 'true' : 'false'
+		);
+		$regex_pattern = '/<button\s+type="submit"[^>]*>.*?<\/button>/i';
+
+		preg_match( $regex_pattern, $product, $input_matches );
+
+		if ( ! empty( $input_matches ) ) {
+			$product = preg_replace( $regex_pattern, $hidden_is_descendent_of_single_product_block_input . $input_matches[0], $product );
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Filter the add to cart message to prevent the Notice from being displayed when the Add to Cart form is a descendent of a Single Product block
+	 * and the Add to Cart button is clicked.
+	 *
+	 * @param string $message Message to be displayed when product is added to the cart.
+	 */
+	public function add_to_cart_message_html_filter( $message ) {
+		// phpcs:ignore
+		if ( isset( $_POST['is-descendent-of-single-product-block'] ) && 'true' == $_POST['is-descendent-of-single-product-block'] ) {
```

Comment:
Should it be `===`?

Reply:
I just changed to use the identity operator instead of the equality operator. Thank you for pointing this out!